From 0000000000000000000000000000000000000000 Sun Jul 17 00:00:00 2001
From: Oliver Kent <support@newwindserver.com>
Date: Sun, Jul 17 2022 13:57:33 -0100
Subject: [PATCH] Add EntityStopTickingEvent

Adds a new event that is called when an entity stops ticking.

diff --git a/src/main/java/io/papermc/paper/event/entity/EntityStopTickingEvent.java b/src/main/java/io/papermc/paper/event/entity/EntityStopTickingEvent.java
index e69de29b..d5990919 100644
--- a/src/main/java/io/papermc/paper/event/entity/EntityStopTickingEvent.java
+++ b/src/main/java/io/papermc/paper/event/entity/EntityStopTickingEvent.java
@@ -0,0 +1,29 @@
+package io.papermc.paper.event.entity;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Called when an entity stops ticking, for example, when it crosses into a non-ticking chunk.
+ */
+public class EntityStopTickingEvent extends EntityEvent {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    public EntityStopTickingEvent(@NotNull final Entity entity) {
+        super(entity);
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}

